### PR TITLE
Add Basic baremetal application example

### PR DESCRIPTION
**Pull request by kegilbert**
Add example page for a starting bare-metal application the baremetal book.

@bulislaw

### CODE REVIEW

**Review by AnotherButler — APPROVED** [review]

File: docs/baremetal_example.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## Bare metal example
+
+To start using the bare metal profile on Mbed OS 5, please see the blinky bare metal example:
```
Comment (iriark01) [inline docs/baremetal_example.md]: Isn't it normally Blinky?

Comment (AnotherButler) [reply docs/baremetal_example.md]: We have a new example that's basically blinky for bare metal; it's different from the standard blinky.

Comment (AnotherButler) [reply docs/baremetal_example.md]: Oh, do you just mean capitalization?

Comment (iriark01) [reply docs/baremetal_example.md]: Yes, capitalisation.

Comment (SenRamakri) [inline docs/baremetal_example.md]: How about changing this sentence to - "To learn about using the bare metal profile on Mbed OS 5,... ". 

Everything else looks great.

**Review by SenRamakri — APPROVED** [review]

**Comment by AnotherButler** [comment]
ping @bulislaw

**Review by bulislaw — APPROVED** [review]

**Pull request merged**